File: rb/features.rb
# Anapioficeandfire SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AnapioficeandfireFeatures
  def self.make_feature(name)
    case name
    when "base"
      AnapioficeandfireBaseFeature.new
    when "ratelimit"
      AnapioficeandfireRatelimitFeature.new
    when "retry"
      AnapioficeandfireRetryFeature.new
    when "test"
      AnapioficeandfireTestFeature.new
    when "timeout"
      AnapioficeandfireTimeoutFeature.new
    else
      AnapioficeandfireBaseFeature.new
    end
  end
end

File: py/features.py
# Anapioficeandfire SDK feature factory

from feature.base_feature import AnapioficeandfireBaseFeature
from feature.test_feature import AnapioficeandfireTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AnapioficeandfireBaseFeature(),
        "test": lambda: AnapioficeandfireTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
